### DISCUSSION TITLE
Ability to setup default PR merge option  per repo/branch

### DISCUSSION THREAD

**Discussion by dantran** [community/community#15349] [Pull Requests]
see https://github.community/t/squash-and-merge-as-default-pull-request-merge-option/1031 for a detailed discussion on the last 4 years.  I just summarize it in one sentence :-)

**Comment by cgriebel** [comment]
This would be a very useful feature for my company. We prefer squash merging but it causes conflicts when we need to merge a hotfix into dev & main outside of our release cycle.

**Comment by vigneshka** [comment]
+1 on this issue. For our normal dev, we want to use squash merging. When we want to deploy changes to the main branch, we want to merge commits from dev to main without squash merging.
